r-package/survlearners/R/surv_fl_grf.R

```diff
@@ -36,31 +36,31 @@
 surv_fl_grf <- function(X, Y, W, D, times, alpha = 0.05, W.hat = NULL, cen.fit = "Kaplan-Meier") {
 
   # IPCW weights
+  Q <- as.numeric(D == 1 | Y > times)    # indicator for uncensored at t0
+  U <- pmin(Y, times)                         # truncated follow-up time by t0
   if (cen.fit == "Kaplan-Meier") {
-    shuffle <- sample(length(Y))
-    kmdat <- data.frame(Y = Y[shuffle], D = D[shuffle])
+    shuffle <- sample(length(U))
+    kmdat <- data.frame(U = U[shuffle], Q = Q[shuffle])
     folds <- cut(seq(1, nrow(kmdat)), breaks = 10, labels = FALSE)
     C.hat <- rep(NA, nrow(kmdat))
     for (z in 1:10) {
       testIndexes <- which(folds == z, arr.ind = TRUE)
       testData <- kmdat[testIndexes, ]
       trainData <- kmdat[-testIndexes, ]
-      c.fit <- survival::survfit(survival::Surv(trainData$Y, 1 - trainData$D) ~ 1)
-      cent <- testData$Y; cent[testData$D == 0] <- times
-      C.hat[testIndexes] <- summary(c.fit, times = cent)$surv
+      c.fit <- survival::survfit(survival::Surv(trainData$U, 1 - trainData$Q) ~ 1)
+      C.hat[testIndexes] <- summary(c.fit, times = testData$U)$surv
     }
     shudat <- data.frame(shuffle, C.hat)
     C.hat <- shudat[order(shuffle), ]$C.hat
   } else if (cen.fit == "survival.forest") {
     c.fit <- grf::survival_forest(cbind(W, X),
-                                  Y,
-                                  1 - D,
+                                  U,
```

Comment:
@CrystalXuR  @erikcs I think this should stay as `Y` rather than `U`.

Reply:
And I think it should stay as `D` too rather than `Q`

in CSF we do:

Fit survival_forest on "raw" data (X, Y, D)

then from this fitted curve pick out P(T > U | X) for censoring weights

then fit next final step cate on complete data as defined by Q